tensorflow_quantum/python/differentiators/linear_combination.py

```diff
@@ -96,16 +96,14 @@ def __init__(self, weights, perturbations):
         self.perturbations = tf.constant(perturbations)
 
     @tf.function
-    def differentiate_analytic(self, programs, symbol_names, symbol_values,
-                               pauli_sums, forward_pass_vals, grad):
+    def get_intermediate_logic(self, programs, symbol_names, symbol_values,
+                               pauli_sums):
+        """Returns copies of the input programs for each perturbed symbol.
 
-        # these get used a lot
-        n_symbols = tf.gather(tf.shape(symbol_names), 0)
+        See base class for Args.
```

Comment:
What is being returned here ? The base class just says what the args are but intermediate logic for each differentiation algorithm could be drastically different and should be documented.

Reply:
Changed the API of `get_intermediate_logic`, the base class now describes both `Args` and `Returns`